compiler/ballerina-lang/src/main/java/io/ballerina/build/ModuleCompilation.java

```diff
@@ -66,9 +66,21 @@
         return new DependencyGraph<>(dependencyIdMap);
     }
 
-    void addModuleDependencies(ModuleId moduleId, Map<ModuleId, Set<ModuleId>> dependencyIdMap) {
+    private void addModuleDependencies(ModuleId moduleId, Map<ModuleId, Set<ModuleId>> dependencyIdMap) {
         Package pkg = packageResolver.getPackage(moduleId.packageId());
         Collection<ModuleId> directDependencies = pkg.moduleDependencyGraph().getDirectDependencies(moduleId);
+
+        ModuleContext moduleContext = pkg.packageContext().moduleContext(moduleId);
```

Comment:
`moduleContext` hides the class level field defined in L41. Can we use other name for this?

Reply:
I was on it.